linode/manager
# test: [M3-7863] - Use `happy-dom` instead of `jsdom` in unit tests

## Description 📝

- Replaces `jsdom` with `happy-dom` 😄 
- I recommend googling `jsdom` vs `happy-dom` to understand the basic differences but here are some extra callouts
  - `happy-dom` should perform better than `jsdom` 🏎️  (Not seeing a huge boost in perf, but it seems to be ~30sec  to 1 min faster in CI)
  - Test stability seems the same (there are no stability regressions with `happy-dom`) 🪨 
  - In my opinion, the main "risk" with `happy-dom` is its lack of maturity although this doesn't seems to have any significant impact on us. This was, for the most part, a drop-in replacement. 👴 
  - `happy-dom` is known to work on the Bun runtime whereas `jsdom` does not work (yet). This doesn't matter at all for us right now, but if we ever consider the Bun runtime in the near future, this change might be nice. 🥟 
  - For whatever reason, `happy-dom` seemed to be more correct with the css that it renders. (In the diff, you'll see some changes to `toHaveStyle` assertions. `happy-dom` seems to be more inline with what the browser does which is good) 🎨 

## Changes Required 💻 
- I had to update some style assertions (`toHaveStyle`)
  - I compared the test output to what I see in the browser and `happy-dom` appears to match what my browser shows, which is a win for `happy-dom`
- I had to update a small number of test assertions to use different selectors. I didn't dig deep into why I had to make the changes I did but all of the changes seemed logical. If you have questions on specific changes, I can comment in more detail
- Some unit tests that tested actions menus never opened the ActionMenu. I had to add a `userEvent.click` in a few tests to open the ActionMenu before making assertions. Not sure how they were passing on `jsdom`
- Had to increase timeout for a Database related test. Don't love this, but not too bad in the grand scheme of things

## How to test 🧪

- Look at diff and make sure changes actually make sense and are not just changes made to satisfy the test 👀 
- Verify the `test-manager` Github Action passes ✅ 
- Run some tests locally and verify they pass
  - This change does not fix timeouts and failures when running a large number of tests locally. This is still an issue. @jdamore-linode mentioned we may be able to increase the global timeout to improve this at some point

## As an Author I have considered 🤔
